### DISCUSSION TITLE
How can I set environment variables for WSL2 instances from the Wezterm config?

### DISCUSSION THREAD

**Discussion by nikfp** [wezterm/wezterm#4395] [Q&A]
### What Operating System(s) are you running on?

Windows, macOS

### Which Wayland compositor or X11 Window manager(s) are you using?

N/A

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

When I run Wezterm in MacOS, I'm able to set environment variables through `config.set_environment_variables` and it works great. I'm able to tie my theme in the terminal, Tmux, and Vim in with the OS setting very easily. 

I also work on a windows machine a lot and I use Wezterm there too, and do my dev work in WSL2. I'd like to get the same functionality with the windows/WSL setup as I have on the Mac, but setting env vars the same way isn't working. I assume I need to do something during startup of the WSL shell to pass that variable in but it's not clear in what I can find in the docs. 

If anyone has any insight please let me know. 

Edit: I should also mention that Wezterm is great. Configuration has been a snap other than this one hangup. Thanks for the awesome tool!

**Comment by wez** [comment]
how are you using WSL with wezterm? WslDomains? Manually running wsl.exe? Something else?

**Reply by nikfp** [reply]
I'm doing this in the config currently. Note that setting the env variable is in a conditional and I have another branch for light as well. 

```lua
config.set_environment_variables = {
    OS_THEME = "Dark"
 }

config.default_domain = "WSL:Ubuntu-20.04"
config.default_prog = { "wsl" }
config.default_cwd = "~"
```

This is straight from the docs for setting the var and starting into the WSL instance. 

Glad to provide any other context you need as well.

**Comment by yerlaser** [comment]
Have you found a solution?
Doesn't work for me either :(

**Comment by kuro337** [comment]
Facing the same issue - having a hard time trying to figure out if this is possible or if it's not possible - have scoured all of the docs and settings.

For example - I need a way to determine on my remote if it was launched by WezTerm or not - would appreciate any feedback @wez

**Comment by drop-stones** [comment]
The [documentation](https://wezfurlong.org/wezterm/config/lua/config/set_environment_variables.html) states that `set_environment_variables` support only `"local"` domain.
Therefore, I think we cannot set environment variables in WslDomain using `set_environment_variables` now.

On the other hand, WSL has a `WSLENV` environment variable to share environment variables with Windows (See https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/#what-are-environment-variables).
To set environment variables on WSL using `WSLENV`, I tried the following configuration:

```
{
  label = "WSL2",
  args = { "wsl" },
  domain = {
    DomainName = "local",
  },
  set_environment_variables = {
    FOO = "bar",
    WSLENV = "FOO",
  }
}
```

I expected the `FOO` environment variable to be set on WSL.
However, the user defined `WSLENV` was not reflected because wezterm overrides the `WSLENV` environment variable automatically.
This behavior is described [here](https://wezfurlong.org/wezterm/config/lua/config/term.html).

It would be great if `set_environment_variables` could be used to set WSL environment variables.
